### PR TITLE
Adds Signal Library Window Op

### PR DESCRIPTION
**Pull request by suleshahid**
First Op for the TFLM Signal Processing Ops library.
Doc linked in bug.

The Window OP is a custom signal processing OP similar to what is found in tf.signal library, but specific for integer (int16) purposes.

You can directly use this as a builtin op via the op resolver as:
* `op_resolver.AddWindow()` 
* or via python as shown in `window_op_test.py`

3 testing options are provided:
* Micro(C++): `bazel run signal/micro/kernels:window_test`
* Tensorflow/Micro(Python): `bazel run python/tflite_micro/signal:window_op_test`
* Makefile(C++): `make -f tensorflow/lite/micro/tools/make/Makefile test_kernel_window_test`  

BUG=[259145369](http://b/259145369)

### CODE REVIEW

File: signal/python/__init__.py
Comment (advaitjain) [inline signal/python/__init__.py:1]: let's move this to `tflite_micro/python/tflite_micro/signal/`

@rkuester: note that we are starting to populate the folder that we expect to be pulled into the pypi package.

File: signal/python/utils/util.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# limitations under the License.
+# ==============================================================================
+"""Python utility functions."""
+import os
+
+import tensorflow as tf
+from tflite_micro.tensorflow.lite.micro.python.interpreter.src import tflm_runtime
+
+
+def get_tflm_interpreter(concrete_function, trackable_obj):
```
Comment (advaitjain) [inline signal/python/utils/util.py:22]: let's make a bug to change the name and find a better location for this function.

File: signal/src/window.cc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "signal/src/window.h"
+
+#include <cstdint>
+
+namespace tflm_signal {
```
Comment (advaitjain) [inline signal/src/window.cc:20]: let's make TODO to remove the namespace later.

File: signal/tflm_signal.bzl
Comment (advaitjain) [inline signal/tflm_signal.bzl:1]: let's have a follow-up PR to change this to be something more generic, something like py_tf_custom_library (worth a design review).

File: signal/tflm_signal.bzl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    Args:
+      name: The name for the python library target build by this rule.
+      srcs: Python source files for the Python library.
+      deps: Dependencies for the Python library.
+      visibility: Visibility for the Python library.
+      cc_op_defs: A list of c++ src files containing REGISTER_OP definitions.
+      cc_op_kernels: A list of c++ targets containing kernels that are used
+          by the Python library.
+    """
+    binary_path = "python/ops"
```
Comment (advaitjain) [inline signal/tflm_signal.bzl:30]: either update for the new location of the python code.

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: signal/BUILD [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+package_group(
+    name = "signal_friends",
+    packages = tflm_signal_friends(),
+)
+
+cc_library(
+    name = "ops_lib",
+    visibility = ["//visibility:public"],
```
[the remark this answers is not in the document]
Comment (suleshahid) [reply signal/BUILD:16]: Done.

File: signal/micro/kernels/Makefile_kernels.inc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+MICROLITE_CC_KERNEL_SRCS += \
```
[the remark this answers is not in the document]
Comment (suleshahid) [reply signal/micro/kernels/Makefile_kernels.inc:1]: Done.

File: signal/micro/kernels/Makefile_tests.inc [hunk 6]
```diff
@@ -0,0 +1,9 @@
+WINDOW_TEST_SRCS := \
```
[the remark this answers is not in the document]
Comment (suleshahid) [reply signal/micro/kernels/Makefile_tests.inc:1]: Done.

File: signal/micro/kernels/register_signal_ops.h [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#ifndef SIGNAL_MICRO_KERNELS_REGISTER_AUDIO_FRONTEND_OPS_H_
+#define SIGNAL_MICRO_KERNELS_REGISTER_AUDIO_FRONTEND_OPS_H_
+
+#include "tensorflow/lite/micro/micro_common.h"
+
+namespace tflite {
+namespace tflm_signal {
+// TODO(b/160234179): Return custom OP registration by value.
+TFLMRegistration* Register_WINDOW();
```
[the remark this answers is not in the document]
Comment (suleshahid) [reply signal/micro/kernels/register_signal_ops.h:9]: Done.

File: signal/python/__init__.py
Comment (suleshahid) [reply signal/python/__init__.py:1]: Done.

File: signal/python/utils/util.py (hunk 1 above)
Comment (suleshahid) [reply signal/python/utils/util.py:22]: Done.

File: signal/tflm_signal.bzl (hunk 3 above)
Comment (suleshahid) [reply signal/tflm_signal.bzl:30]: Done.

File: python/tflite_micro/signal/utils/util.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  converter = tf.lite.TFLiteConverter.from_concrete_functions(
+      [concrete_function], trackable_obj)
+  converter.allow_custom_ops = True
+  tflite_model = converter.convert()
+
+  return tflm_runtime.Interpreter.from_bytes(tflite_model, arena_size=500000)
+
+def load_custom_op(name):
+  return load_library.load_op_library(
+    resource_loader.get_path_to_datafile('../ops/_' + name + '_op.so'))
```
Comment (suleshahid) [inline python/tflite_micro/signal/utils/util.py:41]: This path can be simplified to just ('_' + name + '_op.so') if this function lives in python/tflite_micro/signal/ops/

**Review by advaitjain — APPROVED** [review]

**Pull request merged**